server/service/osquery_utils/queries.go

```diff
@@ -1303,6 +1310,29 @@ func directIngestDiskEncryption(ctx context.Context, logger log.Logger, host *fl
 	return ds.SetOrUpdateHostDisksEncryption(ctx, host.ID, encrypted)
 }
 
+func directIngestDiskEncryptionKey(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string, failed bool) error {
+	if len(rows) == 0 || failed {
+		// assume the extension is not there
+		return nil
+	}
+
+	var key string
+	for i, row := range rows {
+		if strings.Contains(row["line"], "RecoveryKey") {
```

Comment:
Just wondering out loud if this check should be case-insensitive? And any concerns about false-positive like `<key>RecoveryKeyCreationDate</key>` or the likes?